docs/proposals/secret-configmap-file-mode.md

```diff
@@ -0,0 +1,151 @@
+   readable for a secret is not a nice default, we already take care of not
+   writing it to disk, etc. but the file is created world-readable anyways. Such a
+   default change has been done recently: the default was 0444 in kubernetes <= 1.2
+   and is now 0644 in kubernetes >= 1.3 (and the file is not a regular file,
+   it's a symlink now). This change was done here to minimize differences between
+   configmaps and secrets: https://github.com/kubernetes/kubernetes/pull/25285. But
+   doing it again, and changing to something more restrictive (now is 0644 and it
+   should be 0400 to work with ssh and most apps) seems too risky, it's even more
+   restrictive than in k8s 1.2. Specially if there is no way to revert to the old
+   permissions and some use case is broken by this. And if we are adding a way to
+   change it, like in the option above, there is no need to rush changing the
+   default. So I would discard this.
+
+ * We don't want to people be able to change this, at least for now, and the
+   ones who do, suggest that do it as a "postStart" command. This is acceptable
+   if we don't want to change kubernetes core for some reason, although there
+   seem to be valid use cases. But if the user want's to use the "postStart" for
+   something else, then it is more disturbing to do both things (have a script
+   in the docker image that deals with this, but is not probably concern of the
+   project so it's not nice, or specify several commands by using "sh").
+
+# Proposed implementation
+
+The proposed implementation goes with the first alternative: adding a `mode`
+to the API.
+
+There will be a `defaultMode`, type `int`, in `type SecretVolumeSource` and
+`type ConfigMapVolumeSource` and a `mode`, type `int` too, in `type KeyToPath`.
+The mask provided in any of these fields will be ORed with 0777 to disallow
+setting sticky and setuid bits. It's not clear that use case is needed nor
+really understood.
+
+In other words, the fields will look like this (comments might be improved in
+the implementation):
+```
+type SecretVolumeSource struct {
+        // Name of the secret in the pod's namespace to use.
+        SecretName string `json:"secretName,omitempty"`
+        // If unspecified, each key-value pair in the Data field of the referenced
+        // Secret will be projected into the volume as a file whose name is the
+        // key and content is the value. If specified, the listed keys will be
+        // projected into the specified paths, and unlisted keys will not be
+        // present. If a key is specified which is not present in the Secret,
+        // the volume setup will error. Paths must be relative and may not contain
+        // the '..' path or start with '..'.
+        Items       []KeyToPath `json:"items,omitempty"`
+        // Mode bits to use on created files by default
+        DefaultMode int         `json:"defaultMode,omitempty"`
```

Comment:
Ohh, now that I read your comment again I'm not sure if you prefer this field to affect the directory or not.
My idea was that this setting didn't affect directories and only affects files. `fsGroup` can still be used and you usually don't want the same permissions on dirs and files for the reason you said.

So, the `defaultMode`, IMHO, shouldn't affect directories.

What do you think?